### PR TITLE
Add client configuration for Marksman

### PR DESCRIPTION
**Pull request by alanz**
https://github.com/artempyanykh/marksman

Emacs config:

    (use-package markdown-mode
      :after lsp-mode
      :hook (markdown-mode . lsp)
      :config
      (require 'lsp-marksman))

### CODE REVIEW

**Comment by alanz** [comment]
cc @artempyanykh

**Comment by yyoncho** [comment]
can you follow this PR https://github.com/emacs-lsp/lsp-mode/pull/3508/ and add the documentation stuff?

File: lsp-mode.el [hunk 1]
```diff
@@ -841,7 +841,8 @@ Changes take effect only when a new session is started."
                                         (magik-mode . "magik")
                                         (idris-mode . "idris")
                                         (idris2-mode . "idris2")
-                                        (gleam-mode . "gleam"))
+                                        (gleam-mode . "gleam")
+                                        (markdown-mode . "marksman"))
```
Comment (yyoncho) [inline lsp-mode.el:845]: This change is not needed - markdown-mode is mapped to `markdown`.

Comment (alanz) [reply lsp-mode.el:845]: I was wondering about that.  Thanks.

**Review by yyoncho — APPROVED** [review]

**Comment by artempyanykh** [comment]
Wow! Thanks for putting this together @alanz! 💖

**Comment by artempyanykh** [comment]
I found these settings to be useful when working with markdown:
```lisp
(setq-local completion-ignore-case t)
(setq-local lsp-imenu-sort-methods '(kind position name))
```

@yyoncho is there a way to advertise those for marksman LSP? Or perhaps lsp-mode doesn't want to mess with server-specific settings?

**Comment by yyoncho** [comment]
@artempyanykh we do encourage server-specific settings but these seem to be a bit more opinionated. But still, if you want you can create a custom documentation page and add these as a suggestion. If we change these defaults then we need a way to turn off this customization and it ends up very unconventional and not very user-friendly.

**Comment by artempyanykh** [comment]
@yyoncho thanks, yes, this makes sense!

**Pull request merged**